src/DataFramesMeta.jl

```diff
@@ -410,19 +410,39 @@ function orderby(x::AbstractDataFrame, @nospecialize(args...))
 end
 
 function orderby(x::GroupedDataFrame, @nospecialize(args...))
-
     @warn "orderby behavior now returns a `DataFrame` instead of a `GroupedDataFrame`. " *
           "Group the returned data frame to restore old behavior" maxlog = 2
 
     t = DataFrames.select(x, args...; copycols = false, keepkeys = false)
+    @assert nrow(parent(x)) == nrow(t)
     parent(x)[sortperm(t), :]
 end
 
 """
     @orderby(d, i...)
 
-Sort by criteria. When using a grouped data frame, operation
-is performed at the group level.
+Sort by criteria. When given a `GroupedDataFrame`, operation
+is performed at the group level.`@orderby` always returns a fresh
+`DataFrame` regardless of whether the input is a `DataFrame` or a
+`GroupedDataFrame`.
```

Comment:
```suggestion
Sort rows by values in one of several columns or a transformation of columns.
Always returns a fresh `DataFrame` regardless of whether the input is a `DataFrame` or a
`GroupedDataFrame`.
```
(How things work with `GroupedDataFrame` is mentioned below. It also sounds weird that the second sentence would mention `GroupedDataFrame` before even mentioning `DataFrame`.)